data_structures/stacks/stock_span_problem.py

```diff
@@ -8,44 +8,44 @@
 """
 
 
-def calculation_span(price, s):
+def calculation_span(price: list[float]) -> list[float]:
+    """
+    Calculate the span values for a given list of stock prices.
+    Args:
+        price (list): List of stock prices.
+    Returns:
+    >>> price = [10, 4, 5, 90, 120, 80]
+    >>> calculation_span(price)
+    [1.0, 1.0, 2.0, 4.0, 5.0, 6.0]
+    >>> price = [100, 50, 60, 70, 80, 90]
+    >>> calculation_span(price)
+    [1.0, 1.0, 2.0, 3.0, 4.0, 5.0]
+    >>> price = [5, 4, 3, 2, 1]
+    >>> calculation_span(price)
+    [1.0, 1.0, 2.0, 3.0, 4.0]
+    >>> price = [1, 2, 3, 4, 5]
+    >>> calculation_span(price)
+    [1.0, 2.0, 3.0, 4.0, 5.0]
+    >>> price = [10, 20, 30, 40, 50]
+    >>> calculation_span(price)
+    [1.0, 2.0, 3.0, 4.0, 5.0]
```

Comment:
The output does not agree with the results in https://www.geeksforgeeks.org/the-stock-span-problem/
```suggestion
    [1.0, 2.0, 3.0, 4.0, 5.0]
    >>> calculation_span(price=[100, 80, 60, 70, 60, 75, 85])
    [1.0, 1.0, 1.0, 2.0, 1.0, 4.0, 6.0]
    >>> calculation_span(price=[10, 4, 5, 90, 120, 80])
    [1.0, 1.0, 2.0, 4.0, 5.0, 1.0]
```